modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/GetObjectRelationshipEdgeCandidatesMVCResourceCommandTest.java

```diff
@@ -0,0 +1,294 @@
+		// depth is greater than 4
+
+		Assert.assertEquals(
+			_jsonFactory.createJSONArray(
+			).toString(),
+			_getJSONArray(
+				3, objectDefinitionAAAA.getObjectDefinitionId()
+			).toString());
+
+		_objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationshipAAA_AAAA.getObjectRelationshipId());
+
+		// Get object relationship edges candidates where there is a parent who
+		// is not inside a hierarchical structure
+
+		ObjectDefinition objectDefinitionBBB =
+			ObjectDefinitionTestUtil.addObjectDefinition(
+				"BBB", _objectDefinitionLocalService);
+
+		ObjectRelationship objectRelationshipBBB_AAAA =
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				objectDefinitionBBB.getObjectDefinitionId(),
+				objectDefinitionAAAA.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				StringUtil.randomId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		Assert.assertEquals(
+			_jsonFactory.createJSONArray(
+			).put(
+				JSONUtil.put(
+					"label",
+					_getEdgeLabel(
+						objectDefinitionBBB, objectRelationshipBBB_AAAA)
+				).put(
+					"objectRelationshipId",
+					objectRelationshipBBB_AAAA.getObjectRelationshipId()
+				)
+			).toString(),
+			_getJSONArray(
+				0, objectDefinitionAAAA.getObjectDefinitionId()
+			).toString());
+
+		_objectRelationshipLocalService.deleteObjectRelationship(
+			objectRelationshipBBB_AAAA.getObjectRelationshipId());
+
+		// Get object relationship edges candidates where there is a parent who
+		// is inside a hierarchical structure and is the root
+
+		ObjectRelationship objectRelationshipA_AAAA =
+			_objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				objectDefinitionA.getObjectDefinitionId(),
+				objectDefinitionAAAA.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_CASCADE,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				StringUtil.randomId(),
+				ObjectRelationshipConstants.TYPE_ONE_TO_MANY);
+
+		Assert.assertEquals(
+			_jsonFactory.createJSONArray(
+			).put(
+				JSONUtil.put(
+					"ancestors", _jsonFactory.createJSONArray()
+				).put(
+					"label",
+					_getEdgeLabel(objectDefinitionA, objectRelationshipA_AAAA)
+				).put(
+					"objectRelationshipId",
+					objectRelationshipA_AAAA.getObjectRelationshipId()
+				).put(
+					"root", true
+				)
+			).toString(),
+			_getJSONArray(
+				0, objectDefinitionAAAA.getObjectDefinitionId()
+			).toString());
+	}
+
+	private String _getEdgeLabel(
+		ObjectDefinition objectDefinition,
+		ObjectRelationship objectRelationship) {
+
+		return StringUtil.appendParentheticalSuffix(
+			objectDefinition.getLabel(LocaleUtil.getSiteDefault()),
+			objectRelationship.getLabel(LocaleUtil.getSiteDefault()));
+	}
+
+	private JSONArray _getJSONArray(int depth, long objectDefinitionId)
```

Comment:
@MarinhoFeliphe Rename to getObjectRelationshipEdgeCandidates

Reply:
![image](https://github.com/liferay-objects/liferay-portal/assets/42913501/b4a01277-d39b-449b-8358-53dc39606638)

Reply:
@gabrielwas I needed to rename to `getObjectRelationshipEdgeCandidatesJSONArray`

Reply:
[here](https://github.com/liferay-objects/liferay-portal/pull/2431/commits/14dc2efe515de7bb06a4d5a876450f9b2874a958)